src/coprocessor/codec/chunk/column.rs

```diff
@@ -0,0 +1,425 @@
+            Datum::I64(v) => self.append_i64(*v),
+            Datum::U64(v) => self.append_u64(*v),
+            Datum::F64(ref v) => self.append_f64(*v),
+            Datum::Bytes(ref v) => self.append_bytes(v),
+            Datum::Dec(ref v) => self.append_decimal(v),
+            Datum::Dur(ref v) => self.append_duration(v),
+            Datum::Time(ref v) => self.append_time(v),
+            Datum::Json(ref v) => self.append_json(v),
+            _ => Err(box_err!("unsupported datum {:?}", data)),
+        }
+    }
+
+    pub fn new_fixed_column(fixed_len: usize, init_cap: usize) -> Column {
+        Column {
+            fixed_len,
+            data: Vec::with_capacity(fixed_len * init_cap),
+            null_bitmap: Vec::with_capacity(init_cap / 8),
+            ..Default::default()
+        }
+    }
+
+    pub fn new_var_len_column(init_cap: usize) -> Column {
+        let mut offsets = Vec::with_capacity(init_cap + 1);
+        offsets.push(0);
+        Column {
+            var_offsets: offsets,
+            data: Vec::with_capacity(4 * init_cap),
+            null_bitmap: Vec::with_capacity(init_cap / 8),
+            ..Default::default()
+        }
+    }
+
+    #[inline]
+    fn is_fixed(&self) -> bool {
+        self.fixed_len > 0
+    }
+
+    pub fn reset(&mut self) {
+        self.length = 0;
+        self.null_cnt = 0;
+        self.null_bitmap.clear();
+        if !self.var_offsets.is_empty() {
+            // The first offset is always 0, it makes slicing the data easier, we need to keep it.
+            self.var_offsets.truncate(1);
+        }
+        self.data.clear();
+    }
+
+    pub fn is_null(&self, row_idx: usize) -> bool {
+        if let Some(null_byte) = self.null_bitmap.get(row_idx >> 3) {
+            null_byte & (1 << ((row_idx) & 7)) == 0
+        } else {
+            panic!("index out of range");
+        }
+    }
+
+    fn append_null_bitmap(&mut self, on: bool) {
+        let idx = self.length >> 3; // /8
+        if idx >= self.null_bitmap.len() {
+            self.null_bitmap.push(0);
+        }
+        if on {
+            let pos = self.length & 7; // %8
+            self.null_bitmap[idx] |= 1 << pos;
+        } else {
+            self.null_cnt += 1;
+        }
+    }
+
+    pub fn append_null(&mut self) -> Result<()> {
+        self.append_null_bitmap(false);
+        if self.is_fixed() {
+            let len = self.fixed_len + self.data.len();
+            self.data.resize(len, 0);
+        } else {
+            let offset = self.var_offsets[self.length];
+            self.var_offsets.push(offset);
+        }
+        self.length += 1;
+        Ok(())
+    }
+
+    fn finish_append_fixed(&mut self) -> Result<()> {
+        self.append_null_bitmap(true);
+        self.length += 1;
+        self.data.resize(self.length * self.fixed_len, 0);
+        Ok(())
+    }
+
+    pub fn append_i64(&mut self, v: i64) -> Result<()> {
+        self.data.encode_i64_le(v)?;
```

Comment:
Does it produce the same result as this in golang:
```go
374     *(*int64)(unsafe.Pointer(&c.elemBuf[0])) = i
```

Reply:
yes, you can check the `decode_i64_le`  function.